assets/src/hooks/v2/use_simulation_api_response.tsx

```diff
@@ -0,0 +1,161 @@
+import { WidgetData } from "Components/v2/widget";
+import useDriftlessInterval from "Hooks/use_driftless_interval";
+import React, { useEffect, useState } from "react";
+import { ApiResponse } from "./use_api_response";
+
+const MINUTE_IN_MS = 60_000;
+
+interface RawResponse {
+  data: WidgetData | null;
+  force_reload: boolean;
+  disabled: boolean;
+}
+
+type SimulationApiResponse =
+  // The request was successful.
+  {
+    fullPage: ApiResponse;
+    flexZone: ApiResponse;
+  };
+
+const FAILURE_RESPONSE: SimulationApiResponse = {
+  fullPage: { state: "failure" },
+  flexZone: { state: "failure" },
+};
+
+const rawResponseToApiResponse = ({
+  data,
+  disabled,
+}: RawResponse): ApiResponse => {
+  if (disabled) {
+    return { state: "disabled" };
+  } else if (data != null) {
+    return { state: "success", data };
+  } else {
+    return { state: "failure" };
+  }
+};
+
+const rawResponseToSimulationApiResponse = ({
+  full_page,
+  flex_zone,
+}: {
```

Comment:
This function, or some other logic in `useSimulationApiResponse`, needs to also handle the "screen is disabled" and "please reload" responses since it's currently possible for the endpoint to send those back, based on what's in the controller action down below:
```ex
outdated?(screen_id, last_refresh) ->
  json(conn, ScreenData.outdated_response())

disabled?(screen_id) ->
  json(conn, ScreenData.disabled_response())
```

Reply:
Good callout. I think my response is should we care about outdated or disabled responses in these simulations? I'm actually not sure behavior of disabled screens has been brought up yet. Do we show simulations for those at all in Screenplay? We have that in the Screenplay data (although other than re-running the script, we really don't have a way to update that when a change is made in the admin tool), so we could use that to prevent iframes from being present for disabled screens. As for outdated, it kinda seems unnecessary for Screenplay. I don't really see these simulations being looked at long enough for it to get in that state. What do you think?

Reply:
Good question. If we have the data available in screenplay, then it's probably ok to just show "This screen is manually disabled" or something rather than loading up the iframe. If the client is outdated, then it should refresh itself, even if it's a simulation client. You're right that it would probably be a rare case, but it shouldn't be a challenge to implement.